### PR TITLE
copr: fix Max/Min bug when comparing signed and unsigned int64

### LINKED ISSUE #10158: copr bug: Agg func Max/Min got bug when compare signed and unsigned int64
### What version of TiKV are you using?

master

### What operating system and CPU are you using?

Ubuntu 18.04

model name      : AMD Ryzen 5 PRO 4650U with Radeon Graphics
### Steps to reproduce

  drop table t1;
  CREATE TABLE t1  (COL1 bigint(16) UNSIGNED NULL DEFAULT NULL);
  INSERT INTO t1 VALUES (11333449897206147134);
  INSERT INTO t1 VALUES (2420174916247255494);
  INSERT INTO t1 VALUES (3899490809029152765);
  select max(col1), sum(isnull(col1)), count(1) from t1;

### What did you expect?
expect: 11333449897206147134
### What did happened?
but got: 3899490809029152765

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #10158 

Problem Summary: The max function does not consider whether int64 is unsigned when comparing, and always assumes that the input is signed. Therefore, the comparison result of 11333449897206147134 and 3899490809029152765 is problematic.

    drop table t1;
    CREATE TABLE t1 (COL1 bigint(16) UNSIGNED NULL DEFAULT NULL);
    INSERT INTO t1 VALUES (11333449897206147134);
    INSERT INTO t1 VALUES (2420174916247255494);
    INSERT INTO t1 VALUES (3899490809029152765);
    select max(col1), sum(isnull(col1)), count(1) from t1;
    
    Use unistore(Following result is correct):
    
    tidb(localhost:4001) > select max(col1), sum(isnull(col1)), count(1) from t1;
    +----------------------+-------------------+----------+
    | max(col1)            | sum(isnull(col1)) | count(1) |
    +----------------------+-------------------+----------+
    | 11333449897206147134 |                 0 |        3 |
    +----------------------+-------------------+----------+
    1 row in set (0.01 sec)

    use tikv(Following result is wrong):
    
    tidb(localhost:4000) > select max(col1), sum(isnull(col1)), count(1) from t1;
    +---------------------+-------------------+----------+
    | max(col1)           | sum(isnull(col1)) | count(1) |
    +---------------------+-------------------+----------+
    | 3899490809029152765 |                 0 |        3 |
    +---------------------+-------------------+----------+
    1 row in set (0.00 sec)
### What is changed and how it works?

What's Changed: Add a new implementation for Int comparison. We use a const template argument to determinate whether input is unsigned or not.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test: test_aggr_unsigned_signed_int

Side effects: no

### Release note 
- copr: fix Max/Min bug when comparing signed and unsigned int64

### CODE REVIEW

**Comment by guo-shaoge** [comment]
@skyzh PTAL

File: components/tidb_query_aggr/src/impl_max_min.rs [hunk 1]
```diff
@@ -712,6 +787,77 @@ mod tests {
...
+
+        test_integration_util(field_type.tp(), field_type.flag(), &mut columns, &logical_rows, &expected_res);
+        
+        // test signed bigint
+        let mut columns = LazyBatchColumnVec::from(vec![{
+            let mut col = LazyBatchColumn::decoded_with_capacity_and_tp(0, EvalType::Int);
+            col.mut_decoded().push_int(None);
+            col.mut_decoded().push_int(Some(-1));
+            col.mut_decoded().push_int(Some(2420174916247255494));
+            col.mut_decoded().push_int(Some(3899490809029152765));
```
Comment (skyzh) [inline components/tidb_query_aggr/src/impl_max_min.rs:826]: This test makes no sense. It seems that `3899490809029152765` is still in the representable range of both `i64` and `u64`. You could use some primitives like `std::i64::MAX as u64 + 1` to test this.

**Comment by skyzh** [comment]
Please fix format and clippy warnings. And it seems that there's an ICE... Maybe making a new commit could fix this.

File: components/tidb_query_aggr/src/impl_max_min.rs (hunk 1 above)
Comment (guo-shaoge) [reply components/tidb_query_aggr/src/impl_max_min.rs:826]: `-1i64` is `u64::MAX`, so it's lager than 2420174916247255494 and 3899490809029152765.

**Review by skyzh — APPROVED** [review]

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**